[EthanLoww] iP

EthanLoww opened the pull request:
### Sigma Chatbot

> "Just like my Fortnite build battles, I'm building a new future with Sigma. Let’s get those Victory Royales in tech!" 🕹️💥

Sigma Chatbot is about to _gyatt_ its way into your tech setup! This isn’t just any chatbot; it’s got the _IShowSpeed_ energy and is ready to go _fanum tax_ on your tasks. Here’s why Sigma is a total w:

- **Intuitive:** So easy to use, you'll be hitting the griddy in no time
- **Versatile:** Handles all your needs like a true goon cave master
- **Efficient:** Quick and slick, just like Andrew Tate’s finest moves

**To get started:**

1. [Download Sigma here](https://github.com/nus-cs2103-AY2425S1/ip/pull/35#) 🏃‍♂️
2. Install it like a true **_sigma_**
3. Add your tasks and let Sigma handle it with _maximum rizz_
4. Enjoy seamless interactions, because Sigma’s going _full gyatt_ 🚀

**Upcoming Features:**

- [ ] **Advanced Analytics:** For those redpilled insights
- [ ] **Customizable Responses:** Tailored to your based needs
- [ ] **Integration with Other Platforms:** Because we don’t do cringe

Sigma is not only powerful but also **FREE**! Experience the future of chatbot technology with Sigma!

If you are a Java programmer, you can use it to practice Java too. Here’s the main method:
```java
public class Launcher {
    public static void main(String[] args) {
        Application.launch(Main.class, args);
    }
}
```

Get ready to have your tech game _bussin_ and stay tuned for more epic updates. Sigma’s here to bring the _heat_!

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

EthanLoww committed:
Update README.md and add SkibidiSigma.java

EthanLoww committed:
Implement command echo and exit functionality

EthanLoww committed:
Add TaskList.java and update SkibidiSigma.java

EthanLoww committed:
Add Task.java

- Implemented functionality to mark tasks as done and not done.
- Added methods to `Task` class for changing task status.
- Updated `TaskList` class to handle marking tasks as done or not done.
- Modified `SkibidiSigma` class to process 'mark' and 'unmark' commands.
- Enhanced task listing to reflect current status.

EthanLoww committed:
Update Task.java and TaskList.java

-Added toString method to Task

EthanLoww committed:
Add support for ToDos, Deadlines, and Events

- Added `Todo`, `Deadline`, and `Event` classes inheriting from `Task`
to represent different task types.
- Updated `SkibidiSigma` to handle adding and listing tasks of all
types.

EthanLoww committed:
Add support for ToDos, Deadlines, and Events

- Added `Todo`, `Deadline`, and `Event` classes inheriting from `Task`
to represent different task types.
- Updated `SkibidiSigma` to handle adding and listing tasks of all
types.

EthanLoww committed:
Update SkibidiSigma

-Removed logo

EthanLoww committed:
Update SkibidiSigma.java

- Edited Hello message

EthanLoww committed:
Add input and expected output files for text-based UI testing

- Added input.txt to simulate user input for the text-based UI.
- Added EXPECTED.txt with the expected output corresponding to
input.txt.
- Ensured coverage of basic functionalities: adding tasks, marking as
done, unmarking, and listing tasks.

EthanLoww committed:
Add exception handling for invalid commands and empty descriptions

- Introduced `SkibidiSigmaException` for custom error messages.
- Added checks to handle cases where 'todo', 'deadline' and 'event'
commands are used without a description.
- Updated command handling to throw exceptions with appropriate messages
 for empty descriptions and malformed inputs.
- Improved user feedback with clear error messages for incorrect command
 syntax.

EthanLoww committed:
Add delete command handling to SkibidiSigma

- Implemented handleDeleteCommand method to delete tasks from the list.
- Updated main method to include the delete command.
- Switched from using Task[] to ArrayList<Task> for task management.
- Adjusted error handling for invalid task number during deletion.
- Updated relevant parts of the code to maintain task integrity and
reflect changes in the task list.

EthanLoww committed:
Introduce CommandType enum and modularize exception handling

- Created a CommandType enum to standardize command identification
across the application.
- Refactored exceptions by separating them into individual classes for
better maintainability and clarity.
- Enhanced error handling by associating specific exceptions with
corresponding commands.
- Improved code structure and readability by organizing command logic
and exception management.

EthanLoww committed:
Organize classes into packages for improved structure

- Consolidated all classes into appropriate packages to enhance the
organization and maintainability of the codebase.

EthanLoww committed:
Implement UI and Parser classes for improved modularity

- Extracted the UI class to handle user interactions, enhancing code
separation and readability.
- Created the Parser class to centralize command parsing logic, allowing
for cleaner and more maintainable code.
- Improved overall code structure by modularizing UI and command parsing
functionalities.

EthanLoww committed:
Implement task saving and loading

- Save tasks to disk automatically whenever the task list changes.
- Load task data from the disk when the chatbot starts up.
- Create necessary directories if they do not exist.
- Handle cases where the data file does not exist or is corrupted.
- Ensure tasks are saved in a specific format:
  - T | 1 | read book
  - D | 0 | return book | June 6th
  - E | 0 | project meeting | Aug 6th 2-4pm
  - T | 1 | join sports club

EthanLoww committed:
Add Date and Time Parsing for Deadlines

- Enhanced the chatbot to understand and parse dates and times from
input commands.

EthanLoww committed:
Merge branch 'branch-Level-8' into add-gradle-support

EthanLoww committed:
Revert "Merge branch 'branch-Level-8' into add-gradle-support"

This reverts commit ec8c38f27672d334904d8c376f84a3d4be05cd98, reversing
changes made to 81a9c537afe8492675eb253968939364d65db091.

EthanLoww committed:
Reapply "Merge branch 'branch-Level-8' into add-gradle-support"

This reverts commit d8b0b7ac5aba1ea26d404fbdcd65153c0e58bfd9.

EthanLoww committed:
Integrate Gradle Support for Project Automation

- Merged the `add-gradle-support` branch into the master branch.
- Set up Gradle to automate the build process for the Duke project.
- Configured Gradle to build and run the Duke application seamlessly.

EthanLoww committed:
Add JUnit Tests for Non-Trivial Methods

- Implemented JUnit tests for two non-trivial methods across different
classes.
- Added test cases for the `stringify()` and `toString()` methods in the
`Todo` and `Deadline` classes, ensuring proper formatting and status
representation.

EthanLoww committed:
Add JavaDoc comments to classes and methods

- Added JavaDoc comments to all non-private classes and methods.
- Included detailed descriptions for each method, explaining their
purpose, parameters, and exceptions.
- Added comments to non-trivial private methods to enhance code
documentation and readability.

EthanLoww committed:
Tweak code to comply with coding standard

- Updated existing code to adhere to the specified coding standards.
- Ensured consistency with naming conventions, indentation, and
formatting as per the project's coding guidelines.

EthanLoww committed:
Implement task search functionality

- Added `containsKeyword` method to the `Task` class to allow
case-insensitive keyword matching.
- Created `search` method in the `TaskList` class to filter tasks by
the provided keyword.
- Introduced `showSearchedTasks` method in the `Ui` class to display
tasks matching the search criteria.
- Enhanced exception handling with `SigmaEmptyTaskListException` to
handle cases where the task list is empty.

EthanLoww committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/sigma/Sigma.java

EthanLoww committed:
Merge branch 'master' into branch-A-Level9

# Conflicts:
#	src/main/java/sigma/Sigma.java

EthanLoww committed:
Revert "Merge branch 'master' into branch-A-Level9"

This reverts commit 19340481c6005497b97846f3802fa52f25c534f5, reversing
changes made to b747e434062a5152531df7da7178661d187011de.

EthanLoww committed:
Add Javadoc comments to Sigma class

- Documented the main method as the entry point of the application.
- Added descriptions for various methods to clarify their purpose and
functionality.

EthanLoww committed:
Refactor: Apply CheckStyle to detect and resolve coding style violations

- Integrated CheckStyle into the project to ensure adherence to coding
standards.
- Ran CheckStyle and identified areas with coding style violations.
- Refactored code to comply with the project's coding standards.
- Ensured all CheckStyle warnings and errors were resolved.

EthanLoww committed:
Level-10

EthanLoww committed:
Add JavaFX GUI for chatbot

- Implemented basic JavaFX interface for the chatbot.
- Added text input field for user interaction.
- Created a send button for submitting user input.
- Incorporated scrollable area for displaying conversation history.
- Applied basic styling to the GUI components.

Reallyeasy1 commented on src/main/java/sigma/Sigma.java:
Why did you delete javadocs?

EthanLoww committed:
Close window after bye command

EthanLoww committed:
Add assertions to enforce key assumptions in Sigma class

The Sigma class currently lacks assertions to verify critical
assumptions during execution. This can lead to undetected issues if
assumptions are violated, which may cause unexpected behaviour or
errors at runtime.

Introduce `assert` statements in key areas, such as verifying task list
is not empty.

Implement assertions to document and enforce these assumptions, helping
to catch potential issues early in the development process.

This approach improves code reliability and maintains the integrity of
the task management logic throughout the application's lifecycle.

EthanLoww committed:
Add A-CodeQuality

EthanLoww committed:
A-Streams

EthanLoww committed:
Merge pull request #1 from EthanLoww/branch-A-Streams

A-Streams

EthanLoww committed:
Merge pull request #2 from EthanLoww/branch-A-Assertions

Add assertions to enforce key assumptions in Sigma class

EthanLoww committed:
Merge branch 'master' into branch-A-CodeQuality

EthanLoww committed:
Merge pull request #3 from EthanLoww/branch-A-CodeQuality

Add A-CodeQuality

EthanLoww committed:
A-CI

EthanLoww committed:
Support easy updates to existing items

The current implementation requires deleting an item to update its
details. This process is cumbersome and error-prone, particularly when
only minor changes are needed.

Implement functionality to allow users to update existing items directly
without deletion. This enhancement will streamline the process of
modifying details such as the end time of an event, making it more
efficient and user-friendly.

Let's,
* Add functionality to update items directly.
* Ensure that only the specified details are modified, leaving other
attributes unchanged.
* Validate that updates do not inadvertently affect unrelated parts of
the item.

Allowing updates without deletion improves usability and reduces the
risk of data loss or inconsistency. This change aligns with the goal of
making the application more intuitive and efficient for users.

EthanLoww committed:
Remove workflow

EthanLoww committed:
Add workflow

jiayr99 commented on src/main/java/sigma/Main.java:
```diff
@@ -0,0 +1,41 @@
+ * The {@code Main} class serves as the entry point for launching the Sigma GUI
+ * application using JavaFX and FXML. This class initializes the main window
+ * and sets up the scene for user interaction.
+ */
```
Javadoc is very well done and clear

jiayr99 commented on src/main/java/sigma/Main.java:
```diff
@@ -0,0 +1,41 @@
+            AnchorPane ap = fxmlLoader.load();
+            Scene scene = new Scene(ap);
+            stage.setScene(scene);
+            stage.setMinHeight(220);
```
perhaps 220 and 417 can be defined as constants?

jiayr99 commented on src/main/java/sigma/Sigma.java:
```diff
@@ -0,0 +1,422 @@
+    private String handleDeadlineCommand(String userInput) throws SigmaException {
+        try {
+            String[] parts = userInput.split(" /by ");
+            DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd/MM/yy HH:mm");
```
deadline format could maybe be defined as a constant since it is used multiple times?

jiayr99 commented on src/main/java/sigma/Ui.java:
```diff
@@ -0,0 +1,183 @@
+     *
+     * @return A string listing all commands and their descriptions.
+     */
+    public String showHelp() {
```
I like this command that you added as it is unique:+1:

jiayr99 reviewed (commented):
Overall, I think that everything is really well done. Your javadoc is done extremely well and you code had basically no coding violations. Great job!!!

realqijun commented on src/main/java/sigma/Storage.java:
```diff
@@ -0,0 +1,118 @@
+
+        switch (parts[0]) {
+        case "T":
+            task = new Todo(parts[2]);
```
If the file has a line thats just T, this might throw an exception? type shi namsayin

realqijun commented on src/main/java/sigma/task/Event.java:
```diff
@@ -0,0 +1,57 @@
+     */
+    @Override
+    public String stringify() {
+        return String.format("E | %d | %s | %s | %s",
+                this.isDone ? 1 : 0, this.description, this.from.format(FORMATTER), this.to.format(FORMATTER));
```
🤩 Good use of String.format here 🤩

realqijun commented on src/main/resources/images/DaUser.png:
This is insaneeee, you got skibidi taste

realqijun commented on src/main/java/sigma/Sigma.java:
```diff
@@ -0,0 +1,422 @@
+        if (description.isEmpty()) {
+            throw new SigmaMissingArgException(CommandType.EVENT);
+        }
+        if (to.isBefore(from)) {
```
***
Nice handling for valid event times 
***

realqijun commented on src/main/java/sigma/Sigma.java:
```diff
@@ -0,0 +1,422 @@
+            String s = this.ui.showDeletedTask(taskList.getTask(taskNumber), taskList.getSize() - 1);
+            taskList.deleteTask(taskNumber);
+            assert taskList.getSize() == (taskList.getSize() + 1) - 1;
+            return s;
```
I suggest splitting some of the logic here by inserting newlines between them so that beginners like me can read it easier.

realqijun commented on src/main/java/sigma/Sigma.java:
```diff
@@ -0,0 +1,422 @@
+     * @return A confirmation message of the update.
+     * @throws SigmaException If the task number is invalid, or if the update details are incorrect.
+     */
+    public String handleUpdateCommand(String userInput) throws SigmaException {
```
Very cool that you introduced an update task function 🦑🦑🦑🦑
Suggestion: maybe you can also let users know when the task was last updated 😁

realqijun reviewed (commented):
Reviewed the wrong repo my bad gang

snowstopxt commented on src/main/java/sigma/task/Task.java:
```diff
@@ -0,0 +1,74 @@
+    /**
+     * Marks this task as not done.
+     */
+    public void markAsNotDone() {
```
Perhaps using setNotDone will adhere to setters naming conventions for boolean setters :)

snowstopxt commented on src/test/java/sigma/TodoTest.java:
```diff
@@ -0,0 +1,29 @@
+
+public class TodoTest {
+    @Test
+    public void todoTestOne() {
```
Perhaps naming the test addAndMarkTodoTest lets us understand what the test do?

snowstopxt commented on src/main/java/sigma/exception/SigmaFileFormatException.java:
```diff
@@ -0,0 +1,16 @@
+/**
+ * Thrown when the file format is incorrect or when an unknown task type is encountered.
+ */
+public class SigmaFileFormatException extends SigmaException {
```
I like how you named each exception specifically

snowstopxt commented on src/main/java/sigma/Ui.java:
```diff
@@ -0,0 +1,183 @@
+    /**
+     * Displays the welcome message when the application starts.
+     */
+    public String showWelcome() {
```
I like how you named each method for their specific text UI

snowstopxt reviewed (commented):
LGTM. A few nits to fix

EthanLoww committed:
Create unique playful personality for Sigma

- Gave Sigma a more playful and quirky personality.
- Updated phrases to be fun, enthusiastic, and friendly for responding
to commands.

EthanLoww committed:
Update code to adhere to coding standards

EthanLoww committed:
Update gradle.yml

EthanLoww committed:
Add User Guide to README.md

EthanLoww committed:
Merge branch 'master' of https://github.com/EthanLoww/ip

EthanLoww committed:
Update README.md

EthanLoww committed:
Update README.md

EthanLoww committed:
Update README.md
